File: host/c/flash_led.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <unistd.h>
#include <libusb-1.0/libusb.h>

#define VENDOR_ID   0xdead   // Replace with your USB device's vendor ID
#define PRODUCT_ID  0xbeef   // Replace with your USB device's product ID

static bool sendControlTransfer(uint16_t val);

libusb_context* ctx = NULL;
libusb_device_handle* dev_handle = NULL;

int main() {
    // Initialize libusb
    if (libusb_init(&ctx) != 0) {
        fprintf(stderr, "libusb initialization failed\n");
        return 1;
    }

    // Open the USB device using vendor and product ID
    dev_handle = libusb_open_device_with_vid_pid(ctx, VENDOR_ID, PRODUCT_ID);

    if (dev_handle == NULL) {
        fprintf(stderr, "Could not open USB device\n");
        libusb_exit(ctx);
        return 1;
    }

    while(1) {
        if(sendControlTransfer(0x01)) return 1;
        sleep(1);
        if(sendControlTransfer(0x00)) return 1;
        sleep(1);
    }

    // Close the device and exit
    libusb_close(dev_handle);
    libusb_exit(ctx);

    return 0;
}

static bool sendControlTransfer(uint16_t val) {
    int result = libusb_control_transfer(
        dev_handle,
        LIBUSB_REQUEST_TYPE_VENDOR, // Request type
        0x01,                       // Request
        val,                        // Value
        0x00,                       // Index
        NULL,                       // Data to send
        0,                          // Length of data
        1000                        // Timeout (in milliseconds)
    );

    if (result < 0) {
        fprintf(stderr, "Control transfer error: %s\n", libusb_error_name(result));
        libusb_close(dev_handle);
        libusb_exit(ctx);
        return 1;
    }
}

File: host/python/flash_led.py
import usb.core
import sys
import time

dev = usb.core.find(idVendor=0xdead, idProduct=0xbeef)
if(dev == None):
    print("Could not find device")
    sys.exit(255)

while(1):
    # Send a control transfer write to the device to turn on the LED
    dev.ctrl_transfer(  0x40,   # bmRequestType
                        0x01,   # bRequest
                        0x0000, # wValue
                        0x0000, # wIndex
                        0x0000) # Length of data or data

    # Wait for 1 second
    time.sleep(1)

    # Send a control transfer write to the device to turn off the LED
    dev.ctrl_transfer(  0x40,   # bmRequestType
                        0x01,   # bRequest
                        0x0001, # wValue
                        0x0000, # wIndex
                        0x0000) # Length of data or data

    # Wait for 1 second
    time.sleep(1)